src/main.rs

```diff
@@ -145,6 +169,71 @@ fn load_ingredient(path: &Path) -> Result<Ingredient> {
     }
 }
 
+fn configure_sdk(args: &CliArgs) -> Result<()> {
+    let ta = r#"{"trust": { "trust_anchors": replacement_val } }"#;
+    let al = r#"{"trust": { "allowed_list": replacement_val } }"#;
+    let tc = r#"{"trust": { "trust_config": replacement_val } }"#;
+    let vs = r#"{"verify": { "verify_after_sign": replacement_val } }"#;
+
+    let mut enable_trust_checks = false;
+
+    match &args.command {
+        Some(Commands::Trust {
+            trust_anchors,
+            allowed_list,
+            trust_config,
+        }) => {
+            if let Some(trust_list) = &trust_anchors {
+                let data = std::fs::read_to_string(trust_list)?;
+                let replacement_val = serde_json::Value::String(data).to_string(); // escape string
+                let setting = ta.replace("replacement_val", &replacement_val);
+
+                c2pa::settings::load_settings_from_str(&setting, "json")?;
+
+                enable_trust_checks = true;
+            }
+
+            if let Some(allowed_list) = &allowed_list {
+                let data = std::fs::read_to_string(allowed_list)?;
+                let replacement_val = serde_json::Value::String(data).to_string(); // escape string
+                let setting = al.replace("replacement_val", &replacement_val);
+
+                c2pa::settings::load_settings_from_str(&setting, "json")?;
+
+                enable_trust_checks = true;
+            }
+
+            if let Some(trust_config) = &trust_config {
+                let data = std::fs::read_to_string(trust_config)?;
+                let replacement_val = serde_json::Value::String(data).to_string(); // escape string
+                let setting = tc.replace("replacement_val", &replacement_val);
+
+                c2pa::settings::load_settings_from_str(&setting, "json")?;
+
+                enable_trust_checks = true;
+            }
+        }
+        None => {}
+    }
+
+    // if any trust setting is provided enable the trust checks
+    if enable_trust_checks {
+        c2pa::settings::load_settings_from_str(r#"{"verify": { "verify_trust": true} }"#, "json")?;
+    } else {
+        c2pa::settings::load_settings_from_str(r#"{"verify": { "verify_trust": false} }"#, "json")?;
+    }
+
+    // enable or disable verification after signing
+    {
+        let replacement_val = serde_json::Value::Bool(!args.no_signing_verify).to_string();
+        let setting = vs.replace("replacement_val", &replacement_val);
+
+        c2pa::settings::load_settings_from_str(&setting, "json")?;
+    }
```

Comment:
Just curious, why are these lines in their own block? Is it because the docs apply to the whole block? I originally thought one of the lines was making use of the `Drop` trait.

Reply:
It does not have to be.  I had the value optional initially.  BTW, the settings mechanism lets you set the whole settings object or just pieces.  I am doing pieces to allow you configure individual parameters.